base/special/exp.jl

```diff
@@ -114,13 +118,13 @@ function exp{T<:Union{Float32,Float64}}(x::T)
         # scale back
         if k > -significand_bits(T)
             # multiply by 2.0 first to prevent overflow, which helps extends the range
-            k == exponent_max(T) && return y*T(2.0)*T(2.0)^(exponent_max(T) - 1)
+            k == exponent_max(T) && return y * T(2.0) * T(2.0)^(exponent_max(T) - 1)
             twopk = reinterpret(T, rem(exponent_bias(T) + k, fpinttype(T)) << significand_bits(T))
             return y*twopk
         else
             # add significand_bits(T) + 1 to lift the range outside the subnormals
             twopk = reinterpret(T, rem(exponent_bias(T) + significand_bits(T) + 1 + k, fpinttype(T)) << significand_bits(T))
-            return y*twopk*T(2.0)^(-significand_bits(T) - 1)
+            return y * twopk * T(2.0)^(-significand_bits(T) - 1)
```

Comment:
@musm I'm still not sure I quite understand the issue here. All these expressions are still being constant folded. e.g. on this branch:
```
julia> import Base.significand_bits

julia> foo(T) = T(2.0)^(-significand_bits(T) - 1)
foo (generic function with 1 method)

julia> @code_llvm foo(Float64)

define double @julia_foo_65393(i8**) #0 !dbg !5 {
L12:
  ret double 0x3CA0000000000000
}

julia> @code_llvm foo(Float32)

define float @julia_foo_65426(i8**) #0 !dbg !5 {
L12:
  ret float 0x3E70000000000000
}
```